src/ext/java/org/opentripplanner/ext/dataOverlay/GenericDataFile.java

```diff
@@ -0,0 +1,178 @@
+
+            if (time == null) {
+                error = String.format("Missing time variable from %s file", file.getAbsolutePath());
+                return;
+            }
+
+            Variable latitude = netcdfFile.findVariable(davaOverlayConfig.getLatitudeVariable());
+            Variable longitude = netcdfFile.findVariable(davaOverlayConfig.getLongitudeVariable());
+
+            HashMap<IndexVariable, Variable> genVariables = new HashMap<>();
+            for (IndexVariable indexVariable : davaOverlayConfig.getIndexVariables()) {
+                genVariables.put(
+                        indexVariable, netcdfFile.findVariable(indexVariable.getVariable()));
+                //todo check if any vars from the configuration are missing from the file?
+            }
+
+            DateUnit dateUnit = new DateUnit(time.getUnitsString());
+            Date dateOrigin = dateUnit.getDateOrigin();
+
+            if (dateOrigin == null) {
+                error = String.format("Missing origin date from %s file", file.getAbsolutePath());
+                return;
+            }
+
+            if (latitude == null) {
+                error = String.format("Missing latitude variable from %s file",
+                        file.getAbsolutePath()
+                );
+                return;
+            }
+
+            if (longitude == null) {
+                error = String.format("Missing longitude variable from %s file",
+                        file.getAbsolutePath()
+                );
+                return;
+            }
+
+            originDate = OffsetDateTime.ofInstant(dateOrigin.toInstant(), ZoneId.systemDefault());
+            timeArray = time.read();
+            latitudeArray = latitude.read();
+            longitudeArray = longitude.read();
+
+            netcdfDataForVariable = new HashMap<>(genVariables.size());
+            for (Map.Entry<IndexVariable, Variable> genVariable : genVariables.entrySet()) {
+                netcdfDataForVariable.put(
+                        genVariable.getKey().getName(), genVariable.getValue().read());
+            }
+        }
+        catch (Exception e) {
+            error = e.getMessage();
+        }
+    }
+
+    /**
+     * Gets the map of variable names and arrays of corresponding data from .nc file
+     *
+     * @return map of variables names and corresponding grid data arrays
+     */
+    public Map<String, Array> getNetcdfDataForVariable() {
+        return netcdfDataForVariable;
+    }
+
+    /**
+     * Returns whether the file was valid or not
+     *
+     * @return whether the file was valid or not
+     */
+    public boolean isValid() {
+        return error == null;
+    }
+
+    /**
+     * Returns error if one is present in the file, null otherwise
+     *
+     * @return error if one is present in the file.
+     */
+    public String getError() {
+        return error;
+    }
+
+    /**
+     * Returns time array
+     *
+     * @return time array
+     */
```

Comment:
Do not repeat only the method signature in the Java doc, if there is nothing to document skip the doc.